src/deletepy/operations/export_ops.py

```
@@ -0,0 +1,409 @@
+    try:
+        output_dir = os.path.dirname(output_file) if os.path.dirname(output_file) else "."
+        if not os.path.exists(output_dir):
+            raise FileNotFoundError(f"Output directory does not exist: {output_dir}")
+
+        # Test write access
+        test_file = os.path.join(output_dir, ".test_write")
+        with open(test_file, "w") as f:
+            f.write("test")
+        os.remove(test_file)
+    except (PermissionError, OSError) as e:
+        raise PermissionError(f"Cannot write to output file {output_file}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)